.github/workflows/cd.yml

```diff
@@ -0,0 +1,61 @@
+name: CD
+
+on:
+  workflow_dispatch:
+  #   push:
+      # Uncomment and add the necessary branches to enable automatic deployment on AWS
+      # branches:
+        # - main
+
+jobs:
+  deploy:
+    name: Deploy on AWS ECS
+    runs-on: ubuntu-latest
+    environment:
+      name: ${{ github.ref == 'refs/heads/main' && 'production' || github.ref_name }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ vars.AWS_REGION }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build, tag, and push image to Amazon ECR
```

Comment:
Any reason why we do this manually instead of using [docker/build-push-action](https://github.com/docker/build-push-action)? I think you can get the imageID as well using it